doc/requirements/Requirements.md

```diff
@@ -0,0 +1,704 @@
+<why>The reference implementation supports it.</why>
+
+PF **MAY** support constraining the parameter minimum and maximum value.
+<why>To catch user out of valid range errors when changing the parameter
+value.</why>
+
+##### String
+PF **SHOULD** support array of characters.
+<why>Everything that a computer can store fits in an array of characters. It can
+be used as a fallback type if no other matches the parameter.</why>
+
+The array maximum size **MAY** be invariant (immutable).
+<nb>This is what the reference implementation does.</nb>
+
+The API to access the string value **SHOULD** support null terminated character
+array. As it is commonly done in C.
+<why>For easy access from C code.</why>
+
+##### Fix point parameter
+PF **SHOULD** support fix point parameters. I.e. integers divided by a fixed power
+of two.
+<why>The reference implementation supports it.</why>
+
+The API to access the values **SHOULD** respect the Qm.n and UQm.n standards.
+<why>It is the main standard for fix point parameters.</why>
+
+PF **SHOULD** support at least `0 <= m + n <= 31` for a Signed Qm.n and
+`0 <= m + n <= 32` for an Unsigned Qm.n (or "UQm.n").
+<why>The reference implementation supports it.</why>
+<ko>The reference implementation only supports Signed Qn.m</ko>
+
+PF **MAY** support constraining the parameter minimum and maximum value.
+<why>To catch user out of valid range errors when changing the parameter
+value.</why>
+<ko>The reference implementation does not support it</ko>
+
+##### Floating point
+PF **SHOULD** support floating point parameters .
+<why>The reference implementation supports it.</why>
+
+The API to access the values **SHOULD** respect the IEEE 754 standard.
+
+PF **SHOULD** support at least 32 and 64 bit size floats.
+<why>The reference implementation supports it.</why>
+<ko>The reference implementation only supports 32bits</ko>
+
+PF **MAY** support constraining the parameter minimum and maximum value.
+<why>To catch user out of valid range errors when changing the parameter
+value.</why>
+
+##### Bit field following the C ABI
+PF **SHOULD** support 1 or more bit sized integers.
+<why>The reference implementation supports it.</why>
+
+Such parameters called are regrouped in a so called bit parameter block.
+The API to access a bit parameter block **SHOULD** give access to a packed bit
+field.
+<why>The reference implementation supports it.</why>
+
+### Parameter adaptation
+
+#### Philosophy
+Parameters exposed by hardware sometimes need to transform a parameter value.
+For example an hardware integer parameter could have a range 64-128 but it might
+be necessary for upper layer to access in a range 0-100.
+
+This transformation (called parameter adaptation in the rest of the document)
+could be done by the syncer. Nevertheless syncers are supposed to contain only
+business logic and should not be impacted by upper layer needs.
+
+#### Definition
+<dl>
+<dt>Parameter adaptation<dt>
+<dd>
```

Comment:
Added a paragraph:
> For example stocking a parameter with a dynamic type, say either a string or a number could be done with a boolean a string and a number but this could not be pretty print and not memory efficient.

Is that ok :+1: